src/ResourceManager/ContainerInstance/Commands.ContainerInstance/Commands/NewAzureContainerGroupCommand.cs

```diff
@@ -0,0 +1,163 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections;
+using System.Management.Automation;
+using System.Security;
+using Microsoft.Azure.Commands.ContainerInstance.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.ContainerInstance.Models;
+
+namespace Microsoft.Azure.Commands.ContainerInstance
+{
+    /// <summary>
+    /// New-AzureRmContainerGroup
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, ContainerGroupNoun, SupportsShouldProcess = true), OutputType(typeof(PSContainerGroup))]
+    public class NewAzureContainerGroupCommand : ContainerInstanceCmdletBase
+    {
+        [Parameter(
+           Mandatory = true,
+           Position = 0,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+           Mandatory = true,
+           Position = 1,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "The container group name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+           Mandatory = true,
+           HelpMessage = "The container image.")]
+        [ValidateNotNullOrEmpty]
+        public string Image { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The container group Location. Default to the location of the resource group.")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The container OS type. Default: Linux")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            OperatingSystemTypes.Linux,
+            OperatingSystemTypes.Windows,
+            IgnoreCase = true)]
+        public string OsType { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The required CPU cores. Default: 1")]
+        [ValidateNotNullOrEmpty]
+        public double? Cpu { get; set; }
```

Comment:
@yolo3301 is there a range of values that can be provided to this parameter? If so, consider using the `ValidateRange` attribute to limit what values can be provided.

Reply:
Added range